src/movies.js

```diff
@@ -1,25 +1,95 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+/* const getAllDirectors = (moviesArray) => moviesArray.map(movie => movie.director); */
+
+function getAllDirectors(moviesArray) {
+  /* const directors = moviesArray.map(function (movie) {
+    return movie.director;
+  }); */
+  const directors = moviesArray.map((movie) => movie.director);
+  return directors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  const moviesCount = moviesArray.filter(
+    (movie) =>
+      movie.director === "Steven Spielberg" && movie.genre.includes("Drama")
+  ).length;
+  return moviesCount;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  if (moviesArray.length === 0) return 0;
+  const avg =
+    moviesArray
+      .filter((movie) => movie.score !== undefined)
+      .map((movie) => movie.score)
+      .reduce((totalScore, score) => totalScore + score) / moviesArray.length;
+  return Math.round(avg * 100) / 100;
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+  const dramaMovies = moviesArray.filter((movie) =>
+    movie.genre.includes("Drama")
+  );
+  if (dramaMovies.length === 0) return 0;
+  const avg =
+    dramaMovies
+      .filter((movie) => movie.score !== undefined)
+      .map((movie) => movie.score)
+      .reduce((totalScore, score) => totalScore + score) / dramaMovies.length;
+  return Math.round(avg * 100) / 100;
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+  return moviesArray.slice().sort((a, b) => {
+    const yearDiff = a.year - b.year;
+    if (yearDiff === 0) {
+      const orderAlf = a.title.localeCompare(b.title);
+      return orderAlf;
```

Comment:
Esto podemos hacer asi!

```suggestion
      return a.title.localeCompare(b.title);
```